### ISSUE TITLE
Photo feature

### ISSUE THREAD

**Issue by christerdk** [christerdk/Malmofestivalen-for-Android#3]
On start activity there sould be a photo icon. When clicked, the users will go through the following flow:
1. Camera starts.
2. Users takes photo.
3. User is shown a list of current events.
4. User selects event where photo was taken (or nearby, at least).

After selecting the event, the photo will be uploaded to an image server with the following metadata:

1 . Id of schedule. 
2. Timestamp.

(The idea is to show these images later on a site or even live during the festival).

**Comment by mirontoli** [comment]
Sounds interesting. I could try the camear stuff.

**Comment by christerdk** [comment]
I've been looking around for some basic code snippets, just to get over the first hurdles, and I found this: http://snippets.dzone.com/posts/show/8683

Might be a good starting point.

**Comment by christerdk** [comment]
Furthermore, this will be the flow on the server side:
1. Data is received and stored.
2. Mail is sent to group of moderators (me, and others, depending...), which contains a direct approval link.
3. When clicking the link, the moderator is shown a page telling that the image has been approved.
4. When approved, the picture is forwarded to some picture host, preferably someone that supports posting metadata too (here scheduleId etc.).

More or less something like that...

**Comment by mirontoli** [comment]
What kind of picture host do you have in mind? flickr?

**Comment by christerdk** [comment]
That's a possible option, but the Malmö festival web crew spoke very well
about http://www.smugmug.com/ which they use for the site. They said that
the api was very good...

I see the role of that site as being a CDN enabled data store, with good
api and maybe a rss feed generator.

What do you think?

**Comment by FredrikL** [comment]
1. Why not use MediaStore.ACTION_IMAGE_CAPTURE instead of writing our own image capture?
2. Shouldn't we support picking image to upload from gallery too? For those that didn't use the app during the event

**Comment by christerdk** [comment]
@FredrikL 
1. You mean along the lines of http://www.tutorialforandroid.com/2010/10/take-picture-in-android-with.html ? I'd prefer the most standard way to do it, so that sounds good to me (i have no personal experience with image capturing).
2. That's a nice option too! My suggestion is to get the first client case working (start capture + select event + post to server), then we can consider more user flows (such as also capturing a picture from event detail view and, as you suggest, select an image from the gallary). Sounds good? You're welcome to create the idea as an "issue"...

**Comment by christerdk** [comment]
@FredrikL 
Btw, Anatoly and I originally talked about splitting the work, so that he would focus on the app side, and me on the server side. Now that you've joined the team, I'll leave it to you guys to distribute the app work among you, and I'll continue to stay on the server side.

**Comment by FredrikL** [comment]
@christerdk yeah, i used that method in my tumblr app, the important thing is to use the extra_output to get the camera app to write the full resolution image somewhere. Shouldn't be that hard.. i'll familiarize myself with the code a bit more tomorrow.

@mirontoli have you started doing any work on this feature yet ?

**Comment by christerdk** [comment]
Great. I'll slowly start out on the backend stuff, and make a feature branch in my repository. I'll probably start out with just making an void endpoint, that will respond with 200 OK but won't really do anything else for starters. Just so that you guys have something to integrate against.

It there anything about this feature that is not clear? I mean, with regards to the description in the issue and also the follow up comment about what needs to be done on the server? In such case let me know ...

**Comment by christerdk** [comment]
After thinking this over little, I think I will try to get some more validity to this feature, and investigate if anyone has a genuine interest in using the pictures.  I mean, contrary to other features we've had in this project, the idea is only good if someone actually uses the produced images, in my opinion. That is, something along the lines of integrating back to the malmo festival web site (images under artists, or something they'd like to show on big screen during pauses between events). I just think it would be a potential waste of time and creativity, if we get hindered because of unforseen obstacle (technical, political, etc.). If you agree, I'll try to get in touch with some malmo festival people...

Until then, shall we postpone this and find other areas to improve? 

Any thoughts?

**Comment by FredrikL** [comment]
@christerdk sounds good

**Comment by mirontoli** [comment]
@christerdk Sounds good. @FredrikL I haven't started yet. I have very little time from monday to friday :) Next weekend I want to investigate more the issue #7. 

Btw, I am very glad that you joined!
